### DISCUSSION TITLE
Is fetching data in `beforeEnter` and pass props to component is OK?

### DISCUSSION THREAD

**Discussion by dante01yoon** [vuejs/router#1938] [Help and Questions] [answered]
Greetings,

I'm finding good data fetching patterns combine with vue-router.
One approach would be below, fetching data inside beforeEnter and pass props to Component. Is this normal approach?
```ts

const router = VueRouter.createRouter({
  history: VueRouter.createWebHistory(),
  routes: [
    ...,
    {
      path: "/post", 
      component: PostTableView,
      // route navigation guard
      beforeEnter: async (to, from, next) => {
        await new Promise((resolve) => {
          setTimeout(() => {
            resolve({ name: "PostTable"});
          },3000)
        }).then(({name}) => {
          to.meta.name = name;
        })
        next();
      },
      props: route => {
        return {
          name: route.meta.name
        }
      }
    },
    ...
  ]
})
```

Also, I wonder why there is no onBeforeEnter hook in vue-router@v4. (Guess route navigation guard's `beforeEnter` can do that?) but can't find any history on this.

**Comment by dante01yoon** [comment]
I found those and understood why onBeforeEnter not exists.
  - https://github.com/vuejs/router/issues/1433
  - https://github.com/vuejs/router/issues/1517
still can't sure data fetching in beforeEnter is useful

**Comment by posva** [comment, marked as answer]
Writing to `to.meta` is okay, its a fresh object on each navigation

**Reply by dante01yoon** [reply]
I see! thanks a lot!
